angr/analyses/decompiler/optimization_passes/return_duplicator_base.py

```diff
@@ -143,14 +143,27 @@ def _analyze_core(self, graph: networkx.DiGraph) -> bool:
         self._supergraph = to_ail_supergraph(graph)
         for region_head, (in_edges, region) in endnode_regions.items():
             is_single_const_ret_region = self._is_simple_return_graph(region)
+            dup_pred_nodes = []
+            # duplicate the entire region if at least (N-2) in-edges for the region head is deemed should be duplicated.
```

Comment:
This should be a good mitigation on over dup of regions. I'll open a new issue if it is bad.